### ISSUE TITLE
Preserve titleless feed entries

### ISSUE THREAD

**Issue by Krinkle** [marienfressinaud/FreshRSS#4946]
**Describe the bug**
Titleless feed entries are currently normalised into entries that mask the content behind a generic URL as the title.

This primarly affects microblog feeds such as from Mastodon (and other ActivityPub-based systems in the Fediverse), Twitter, Micro.blog, and likely WordPress in the future (announced today).

**Expected behavior**
Items that don't have a title in the source, should not have a title in the proxied output either. 

**Screenshots**
Below is how NetNewsWire renders the http://scripting.com/ blog directly (without sync), and the same feed when synced through FreshRSS:

| Direct | Synced | Direct | Synced
|--|--|--|--
| <img height="300" alt="" src="https://user-images.githubusercontent.com/156867/208168857-b8ea37f6-38b8-4acc-b659-470cd16d3529.PNG">  | <img height="300" alt="" src="https://user-images.githubusercontent.com/156867/208169652-bd267bb4-5eac-4ff4-98a7-77c82f667b0a.PNG"> | <img height="300" alt="" src="https://user-images.githubusercontent.com/156867/208168870-8e61cab6-d459-4642-9092-01ccc5d9d907.PNG"> | <img height="300" alt="" src="https://user-images.githubusercontent.com/156867/208168956-58f9befb-299c-4997-8ec6-37a10170632f.PNG">

| Direct | Synced
|--|--
| <img height="400" alt="" src="https://user-images.githubusercontent.com/156867/208169134-0b8771f1-40eb-4039-b697-8b6914ea0d0a.png"> | <img height="400" alt="" src="https://user-images.githubusercontent.com/156867/208169356-a0d02a73-26fc-4259-bf8e-7e6c9e680b2d.png">

**To Reproduce**
1. Subscribe to http://scripting.com/rss.xml in FreshRSS.
2. In the default FreshRSS user interface, navigate to the feed and click its feed icon, which points to something like `/i/?a=rss&get=…&rid=…&hours=168`.

**Environment information:**
- FreshRSS version:  1.20.0 
- PHP version: PHP 7.4

**Additional context**

This is **not** a feature request to redesign the FreshRSS's frontend to optimise for titleless posts. This is what it looks like today:
<img height="200" alt="Screenshot" src="https://user-images.githubusercontent.com/156867/208172085-4b693ed3-90a0-4759-a5b0-54de14e5910f.png">

It's totally fair for each frontend to specialise in a certain kind of frontend, or to deprioritise a potentially complex redesign. Some frontends take their `[title] \n [1-line preview of content]` layout and render titleless entries as `[2-line preview of content]`. Or, a layout like `[title] \n [all content]` might omit the title, or use a linked human-readable date as title in the expanded view. Any of those choices are reasonable. FreshRSS's frontend is an understandable trade-off.

Where it becomes difficult, is when the design choices and feature priorities of one frontend become fixed into the feed that is proxied to an entirely different frontend (e.g. NetNewsWire, as shown in the first set of screenshots).

This might present another reason to consider limiting feed normalizations to first-party calls from the UI. See also https://github.com/FreshRSS/FreshRSS/issues/4481 and https://github.com/FreshRSS/FreshRSS/issues/4104, which are other bugs that may or may not be a priority for FreshRSS, but could have been limited to the UI instead of affecting other apps as well.

See also https://github.com/scripting/titlelessFeedsHowto and http://scripting.com/2022/12/16.html by @scripting.

**Comment by math-GH** [comment]
I would not call it a bug.

It is a feature to use the URL as title, if not set by the feed,

scripting.com sets for some articles a title.

See: 
![grafik](https://user-images.githubusercontent.com/1645099/208770311-80e21bc6-aabe-45c7-85f3-9ab4c108b09a.png)

I recommend to update your FreshRSS instance to 1.20.2, that fix a security issue.

**Comment by jeffehobbs** [comment]
I agree with @math-GH. Feeds should fall back on the smallest amount of data that has been specified. If that's a URL, or a URL path, then that's what is available.

**Comment by Frenzie** [comment]
Note that this is a feature request for the API, possibly in the form of a setting or a query parameter for clients, though at least for Mastodon-type content the NetNewsWire style rendering is also nicer tout court. I think it's a good idea.

**Comment by Inverle** [comment]
Is this fixed by https://github.com/FreshRSS/FreshRSS/pull/6240?
